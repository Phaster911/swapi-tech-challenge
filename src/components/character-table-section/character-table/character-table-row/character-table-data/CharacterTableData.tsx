import styled from "styled-components";
import { controlBorderRadius, QUERIES } from "../../../../../style-constants";
import { ReactNode } from "react";

const StyledTableData = styled.td`
  --radius: calc(${controlBorderRadius} * 2);
  padding: 0.5rem;
  display: grid;
  grid-template-columns: 15ch auto;
  gap: 0.5rem;
  text-align: left;

  &:first-child {
    border-top-left-radius: var(--radius);
    border-top-right-radius: var(--radius);
  }
  &:last-child {
    border-bottom-left-radius: var(--radius);
    border-bottom-right-radius: var(--radius);
  }

  &::before {
    content: attr(data-cell) ": ";
    text-transform: capitalize;
    font-weight: 700;
  }

  @media ${QUERIES.tabletAndUp} {
    display: revert;
    padding: 1rem 0.5rem;

    &::before {
      display: none;
    }

    &:first-child {
      border-top-left-radius: var(--radius);
      border-bottom-left-radius: var(--radius);
      border-top-right-radius: 0;
    }
    &:last-child {
      border-top-right-radius: var(--radius);
      border-bottom-right-radius: var(--radius);
      border-bottom-left-radius: 0;
    }
  }
`;

export type CharacterTableDataProps = {
  cellName: string;
  children: ReactNode;
};

export default function CharacterTableData({
  cellName,
  children,
}: CharacterTableDataProps) {
  return <StyledTableData data-cell={cellName}>{children}</StyledTableData>;
}
