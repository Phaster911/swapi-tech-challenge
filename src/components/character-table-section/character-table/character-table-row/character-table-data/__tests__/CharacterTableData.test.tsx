import { render } from "@testing-library/react";
import CharacterTableData from "../CharacterTableData";

describe("<CharacterTableData />", () => {
  it("renders the component", () => {
    const { container } = render(
      <table>
        <tbody>
          <tr>
            <CharacterTableData cellName="field">ABC</CharacterTableData>
          </tr>
        </tbody>
      </table>
    );
    expect(container).toMatchSnapshot();
  });
});
