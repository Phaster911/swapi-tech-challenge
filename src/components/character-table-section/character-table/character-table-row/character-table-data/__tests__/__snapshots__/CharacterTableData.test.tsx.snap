// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CharacterTableData /> > renders the component 1`] = `
.c0 {
  --radius: calc(5px * 2);
  padding: 0.5rem;
  display: grid;
  grid-template-columns: 15ch auto;
  gap: 0.5rem;
  text-align: left;
}

.c0:first-child {
  border-top-left-radius: var(--radius);
  border-top-right-radius: var(--radius);
}

.c0:last-child {
  border-bottom-left-radius: var(--radius);
  border-bottom-right-radius: var(--radius);
}

.c0::before {
  content: attr(data-cell) ": ";
  text-transform: capitalize;
  font-weight: 700;
}

@media (min-width: 750px) {
  .c0 {
    display: revert;
    padding: 1rem 0.5rem;
  }

  .c0::before {
    display: none;
  }

  .c0:first-child {
    border-top-left-radius: var(--radius);
    border-bottom-left-radius: var(--radius);
    border-top-right-radius: 0;
  }

  .c0:last-child {
    border-top-right-radius: var(--radius);
    border-bottom-right-radius: var(--radius);
    border-bottom-left-radius: 0;
  }
}

<div>
  <table>
    <tbody>
      <tr>
        <td
          class="c0"
          data-cell="field"
        >
          ABC
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
